### PR TITLE
Fixes #7126: Add a new system variable for the \"rudder_directives\" bundle

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/7126

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/cfclerk/services/impl/Cf3PromisesFileWriterServiceImpl.scala [hunk 1]
```diff
@@ -347,26 +357,21 @@ class Cf3PromisesFileWriterServiceImpl(
...
       } else {
         techniqueBundles
       }
     }
 
+    List(
+      SystemVariable(systemVariableSpecService.get("INPUTLIST"), Seq(inputs.distinct.mkString("\"", "\",\"", "\"")))
+    , SystemVariable(systemVariableSpecService.get("BUNDLELIST"), Seq(bundleSeq.map( _._2.name).mkString(", ", ", ", "")))
+    , SystemVariable(systemVariableSpecService.get("RUDDER_DIRECTIVES_SEQUENCE"),
+        Seq(bundleSeq.map { case (promisee, bundle) => s""" "${promisee} usebundle => ${bundle.name};" """}.mkString( "\n"))
```
Comment (jooooooon) [inline rudder-core/src/main/scala/com/normation/cfclerk/services/impl/Cf3PromisesFileWriterServiceImpl.scala]: This should be promiser, not promisee. Promisee is a whole different concept in CFEngine. See http://syslog.me/2013/05/21/the-value-of-the-promisee-in-cfengine-promises/.

Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/cfclerk/services/impl/Cf3PromisesFileWriterServiceImpl.scala]: this should indeed be called promiser

Plus, you have invalid quoting, the line should probably be
"${promiser}" usebundle => ${bundle.name};

**Comment by fanf** [comment]
PR updated

File: rudder-core/src/main/scala/com/normation/cfclerk/services/impl/Cf3PromisesFileWriterServiceImpl.scala [hunk 2]
```diff
@@ -312,27 +312,36 @@ class Cf3PromisesFileWriterServiceImpl(
...
-
-
-    val bundleSeq: Seq[Bundle] = techniques.flatMap { technique =>
+    val bundleSeq: Seq[(Technique, String, Bundle)] = techniques.flatMap { case BundleTechnique(technique, promisee) =>
       // We need to remove zero-length bundle name from the bundlesequence (like, if there is no ncf bundles to call)
       // to avoid having two successives commas in the bundlesequence
       val techniqueBundles = technique.bundlesequence.flatMap { bundle =>
         if(bundle.name.trim.size > 0) {
-          Some(bundle)
+          Some((technique, promisee, bundle))
```
Comment (jooooooon) [inline rudder-core/src/main/scala/com/normation/cfclerk/services/impl/Cf3PromisesFileWriterServiceImpl.scala]: This should still be promise**r**

File: rudder-core/src/main/scala/com/normation/cfclerk/services/impl/Cf3PromisesFileWriterServiceImpl.scala [hunk 3]
```diff
@@ -312,27 +312,36 @@ class Cf3PromisesFileWriterServiceImpl(
...
+      bt.technique.templates.collect { case template if(template.included) => (bt.technique, template.outPath) }
     }
 
-    // Compute the correct bundlesequence
-    // ncf technique must call before-hand a bundle to register which ncf technique is being called
-    val NCF_REPORT_DEFINITION_BUNDLE_NAME = "current_technique_report_info"
-
-
-    val bundleSeq: Seq[Bundle] = techniques.flatMap { technique =>
+    val bundleSeq: Seq[(Technique, String, Bundle)] = techniques.flatMap { case BundleTechnique(technique, promisee) =>
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/cfclerk/services/impl/Cf3PromisesFileWriterServiceImpl.scala]: and promiser here also

**Comment by fanf** [comment]
PR updated

**Comment by fanf** [comment]
PR updated

**Pull request merged**